sqlmesh/core/plan/definition.py

```diff
@@ -253,14 +257,26 @@ def environment(self) -> Environment:
             if self.is_dev and self.environment_ttl is not None
             else None
         )
+
+        snapshots = [s.table_info for s in self.snapshots]
+        promoted_snapshot_ids = None
+        if self.is_dev and not self.promote_all:
+            names_to_promote = {
+                *self.context_diff.promoted_model_names,
```

Comment:
i'm not sure i understand this, what is promoted_model_names used for, why do we need it? why do we need to store in it the state sync?

Reply:
We need to track which models were promoted in the target environment during the previous plan application, so that we don't wipe them out while applying subsequent changes to the same environment.

Reply:
i see now. promoted_model_names ensures that you replace model names that were previously created

Reply:
Previously promoted, yes!